plugins/net_plugin/net_plugin.cpp

```diff
@@ -2255,7 +2255,7 @@ namespace eosio {
 
    void net_plugin_impl::emplace_connection(connection_ptr& conn) {
       auto itr = connections.emplace(conn).first;
-      conn->bmi_modify = [&](auto&& func) { connections.modify(itr, func); };
+      conn->bmi_modify = [&itr](auto&& func) { my_impl->connections.modify(itr, func); };
```

Comment:
Nope.
I mean the following change:
```
void set_state(connection_state s) {
￼          my_impl->connections.modify(itr, { this->state = s; });
￼}
```

Reply:
@afalaleev Store itr in connection class?

Reply:
@afalaleev There are some problems to do as you proposing:
1. Which type should have `itr` field in connection class? It will be very complex iterator, like this:
```
      boost::multi_index::detail::bidir_node_iterator<boost::multi_index::detail::ordered_index_node<boost::multi_index::detail::null_augment_policy,
            boost::multi_index::detail::ordered_index_node<boost::multi_index::detail::null_augment_policy,
            boost::multi_index::detail::ordered_index_node<boost::multi_index::detail::null_augment_policy,
            boost::multi_index::detail::ordered_index_node<boost::multi_index::detail::null_augment_policy,
            boost::multi_index::detail::index_node_base<connection_ptr,
            std::allocator<connection_ptr > > > > > > > my_itr;
```
2. `net_plugin_impl` in this `my_impl` is incomplete type so we cannot access `connections` member.